### PR TITLE
Azure AD and PA VPN Fusion Detection

### PR DESCRIPTION
**Pull request by petebryan**
Created new detection to correlate multiple failed AAD logins from a single IP with a successful login to a Palo Alto VPN from the same IP within the same time period.

### CODE REVIEW

File: Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      //Parse out the logon source IP from the Message field to match on
+      | extend SourceIP = extract("Login from: ([^,]+)", 1, Message) 
+      | where SourceIP in (suspicious_signins)
+      | extend Reason = "Multiple failed AAD logins from SourceIP"
+      //Parse out other useful information from Message field
+      | extend User = extract("User name: ([^,]+)", 1, Message) 
+      | extend ClientOS = extract("Client OS version: ([^,\"]+)", 1, Message)
+      | extend Location = extract("Source region: ([^,]{2})",1, Message)
+      | project TimeGenerated, Reason, SourceIP, User, ClientOS, Location, Message, DeviceName, ReceiptTime, DeviceVendor, DeviceEventClassID, Computer, FileName
+      | extend AccountCustomEntity = User, IPCustomEntity = SourceIP
```
Comment (shainw) [inline Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml]: add in timestamp = TimeGenerated and HostCustomEntity = DeviceName, we have been using this for Common Security Log as there is no host, but we could correlate events on a single device if needed

File: Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+tactics:
+  - InitialAccess
+  - CredentialAccess
+relevantTechniques:
+  - T1078
+  - T1110
+
+query: |
+  //Set a threshold of failed AAD signins from an IP address within 1 day above which we want to deem those logins suspicious.
+  let signin_threshold = 5;
```
Comment (shainw) [inline Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml]: set the timeframe at top - 
let timeframe = 1d; 
and then put timeframe in as the variable where you have 1d for example
| where TimeGenerated >= ago(timeframe)

File: Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      | where IPAddress != "127.0.0.1"
+      | summarize count() by IPAddress
+      | where count_ >  signin_threshold
+      | summarize makelist(IPAddress);
+  //See if any of those IPs have sucessfully logged into PA VPNs during the same timeperiod
+  CommonSecurityLog
+      | where TimeGenerated > ago(1d)
+      //Select only PA VPN sucessful logons
+      | where DeviceVendor == "Palo Alto Networks"
+      | where DeviceEventClassID == "globalprotect"
```
Comment (shainw) [inline Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml]: maybe make this an AND - | where DeviceVendor == "Palo Alto Networks" and DeviceEventClassID == "globalprotect"
I believe this is more efficient.

File: Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml (hunk 1 above)
Comment (petebryan) [reply Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml]: Done

File: Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml (hunk 2 above)
Comment (petebryan) [reply Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml]: Done

File: Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml (hunk 3 above)
Comment (petebryan) [reply Detections/MultipleDataSources/AAD_PAVPN_Correlation.yaml]: Done

**Pull request merged**